uaclient/util.py

```diff
@@ -416,16 +418,27 @@ def _subp(
     @raises subprocess.TimeoutError when timeout specified and the command
         exceeds that number of seconds.
     """
+    if data is None:
+        # using devnull assures any reads get null, rather
+        # than possibly waiting on input.
+        stdin = open(os.devnull)
+    else:
+        data = data if isinstance(data, bytes) else data.encode("utf-8")
```

Comment:
We shouldn't need to do this; could we just accept `bytes` and place the onus on callers to encode their data appropriately?

Reply:
good pt. irrelevant now with the content revert